graphql_compiler/compiler/helpers.py

```diff
@@ -237,6 +237,45 @@ def get_location_name(self):
         """Return a tuple of a unique name of the location, and the current field name (or None)."""
         raise NotImplementedError()
 
+    @abstractmethod
+    def _compare(self, other):
```

Comment:
How about `_check_if_object_of_same_type_is_smaller` or something similarly more descriptive than `_compare`? Currently, it doesn't communicate that the other object is required to be of the same type as `self`.